CMakeLists.txt

```diff
@@ -60,74 +60,92 @@ if( ENABLE_TESTING )
     find_package( test-drive REQUIRED )
 endif()
 
+# Optionally use PackageCompiler.jl to build standalone libtrixi.so
+option( USE_PACKAGE_COMPILER "Build standalone libtrixi.so using PackageCompiler.jl" )
+if( USE_PACKAGE_COMPILER )
+    if ( NOT DEFINED JULIA_PROJECT_PATH )
+        message( FATAL_ERROR "JULIA_PROJECT_PATH needs to be set for PackageCompiler.jl.")
+    endif()
 
+    # no sources
+    add_library( ${PROJECT_NAME} INTERFACE build-pc/lib/libtrixi.so )
+    add_custom_command( OUTPUT build-pc/lib/libtrixi.so
```

Comment:
Should we add other files to the output as well, e.g.,
```
build-pc/include/trixi.h
build-pc/share/julia/LIBTRIXI_VERSION
build-pc/share/julia/LocalPreferences.toml
build-pc/share/julia/Project.toml
```
?

I think it is probably too hard/not useful enough to also include the other outputs (such as all the libraries that are bundled), since they are just copy-pasted into the bundle.

Reply:
Actually I do not know (yet) as I do not really understand what is going on in cmake here (yet).

My intention was to define a target similar to what we have when building `libtrixi.so` from sources via the usual C toolchain. 

Installing could work by just copying the share, lib, include directories explicitly. 

Linking the examples so far works by 
```
target_link_libraries(
        ${TARGET_NAME}
        ${PROJECT_NAME}
    )
```
and ideally this (i.e. just specifying `${PROJECT_NAME}` for the library) could be reused.

Reply:
OK, makes sense. Let's keep it as it is for now and (Maybe) reconsider this in the future. I think adding the other missing functionality (building examples etc) is much more important.